dpnp/dpnp_iface_statistics.py

```diff
@@ -506,7 +550,30 @@ def _run_native_sliding_dot_product1d(a, v, l_pad, r_pad):
     return out
 
 
-def correlate(a, v, mode="valid"):
+def _convolve_fft(a, v, l_pad, r_pad, rtype):
+    assert a.size >= v.size
+    assert l_pad < v.size
```

Comment:
These assertions should be turned to checks that raise exceptions if unmet. 

This is because Python interpreter allows to turn off assertion checks with `-O` option:

```
(dev_dpctl) opavlyk@opavlyk-mobl:~/repos/dpctl$ python -O -c "assert False"
(dev_dpctl) opavlyk@opavlyk-mobl:~/repos/dpctl$ python  -c "assert False"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError
```